src/App.js

```diff
@@ -1,24 +1,35 @@
 import React, { Component } from 'react';
 import Area from './movie-area.js';
 import Header from './header.js';
-import movieData from './mock-data.js';
 import Details from './movie-details.js';
+import { getMovie, getVideo } from './api-calls';
 
 class App extends Component {
   constructor() {
     super();
     this.state = {
       movies: [],
       currentMovie: null,
+      currentVideo: null,
+      error: '',
     };
   }
 
   componentDidMount() {
-    this.setState({movies: movieData.movies})
+    getMovie()
+    .then(data => {
+      this.setState({movies: data.movies})
+    })
+    .catch(error => this.setState({ error: 'Something went wrong!'}))
   }
 
   showMovieDetails = event => {
     this.setState({currentMovie: event.target.id})
+    getVideo(event.target.id)
+    .then(video => {
+      this.setState({currentVideo: video})
+    })
+    .catch(error => this.setState({ error: 'Video is unavailable'}))
```

Comment:
Now that we have this waiting for further functionality we can plan how we'd like to display the videos. Perhaps we could display the first video of the videos array, and have buttons on left and right to iterate through the videos array by one index so user can click through all videos associated with a movie.